### PR TITLE
Deploy to other remotes more easily

### PR DESCRIPTION
**Pull request by jonsmithers**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -131,7 +131,9 @@ gulp.task('lint', function () {
 gulp.task('gh-deploy', ['switch-to-src','default'], () => {
     return gulp.src(deployDir+'/**')
         .pipe(ghPages({
-            remote: 'origin',
+            // uncomment one of these to deploy elsewhere
+            // remoteUrl: 'https://github.com/jonsmithers/order-splitter.git',
```
Comment (dumbbillyhardy) [inline gulpfile.js]: what do you think about reading this from a file that isn't staged?

File: gulpfile.js [hunk 2]
```diff
@@ -128,10 +131,36 @@ gulp.task('lint', function () {
...
+            }
+        });
+    } else {
+        remoteUrl = argv.remoteUrl; // add to global scope
+        return Promise.resolve();
+    }
+});
+gulp.task('gh-deploy-helper', () => {
     return gulp.src(deployDir+'/**')
         .pipe(ghPages({
```
Comment (dumbbillyhardy) [inline gulpfile.js:162]: could we also support specifying the remote name. Maybe the variable we pass to this task could be the entire object passed into ghPages?

**Review by dumbbillyhardy — DISMISSED** [review]

File: gulpfile.js (hunk 2 above)
Comment (jonsmithers) [reply gulpfile.js:162]: By "remote name" are you referring to the [`origin`](https://www.npmjs.com/package/gulp-gh-pages#optionsorigin) option? I've never tried but I imagine you specify _either_ `remoteUrl` _or_ `origin`, not both?

Comment (dumbbillyhardy) [reply gulpfile.js:162]: I believe so

Comment (jonsmithers) [reply gulpfile.js:162]: For the `origin` argument, if I use anything other than "origin", I get this error:

```
[07:32:50] Error: Command failed: git push --set-upstream not-origin test-gh-pages
fatal: 'not-origin' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Comment (jonsmithers) [reply gulpfile.js:162]: If you `cd` into `/.publish` and run `git remote -vv`, you'll notice that there's only 1 remote: "origin". Not sure why they bothered making this an argument.

**Comment by jonsmithers** [comment]
It seems we'd have to programmatically determine the remoteUrl from the remote name ourselves. I'd say that's a task for another pull request.

**Pull request merged**